kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -493,4 +496,26 @@ default void removeOwnerReference(HasMetadata owner) {
   default Optional<ObjectMeta> optionalMetadata() {
     return Optional.ofNullable(getMetadata());
   }
+
+  static <T extends HasMetadata & Namespaced> T emptyCopyOf(T original, Supplier<T> constructor) {
+    final T t = constructor.get();
+    final ObjectMeta metadata = Objects.requireNonNull(original.getMetadata(), REQUIRES_NON_NULL_METADATA);
+    final String name = Objects.requireNonNull(metadata.getName(), REQUIRES_NON_NULL_NAME);
+    final String ns = Objects.requireNonNull(metadata.getNamespace(), REQUIRES_NON_NULL_NAMESPACE);
+    t.setMetadata(new ObjectMetaBuilder()
+        .withName(name)
+        .withNamespace(ns)
+        .build());
+    return t;
+  }
+
+  static <T extends HasMetadata> T emptyClusterWideCopyOf(T original, Supplier<T> constructor) {
```

Comment:
Umm, sorry for the naive question. Why is the visibility of this method kept to the default scope?

Reply:
Because this is on an interface so it is by default public.